ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/precompiles/privacy/PrivacyPrecompiledContract.java

```diff
@@ -181,14 +185,23 @@ public PrecompileContractResult computePrecompile(
         processPrivateTransaction(
             messageFrame, privateTransaction, privacyGroupId, privateWorldStateUpdater);
 
+    if (!result.isSuccessful() && incrementPrivateNonce) {
+      privateWorldStateUpdater.commitPrivateNonce();
+      disposablePrivateState.persist(null);
+
+      storePrivateMetadata(
```

Comment:
the reason for which I'm calling the storePrivateMetadata here, which is storing also  the metadata of the transaction even if is null, is because is not enough only to persist the new nonce on the trieState. In order for the nonce to be available for next iteration the lastRootHash needs to be updated with the new state in privateWorldStateArchive, and by calling only  updatePrivacyGroupHeadBlockMap would have create incomplete states, resulting  validation failing.